src/actions/user.profile.action.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import axios from 'axios';
+export const UPDATE_USER_PROFILE = 'UPDATE_USER_PROFILE';
+export const GET_USER_PROFILE = 'GET_USER_PROFILE';
+export const UPLOAD_IMAGE_SUCCESS = 'UPLOAD_IMAGE_SUCCESS';
+export const UPLOAD_IMAGE_FAIL = 'UPLOAD_IMAGE_FAIL';
+export const CHANGE_STATE = 'CHANGE_STATE';
+export const HTTP_REQUEST_START = 'HTTP_REQUEST_START';
+export const CANCEL_USER_UPDATE = 'CANCEL_USER_UPDATE';
+export const ERROR_UPLOAD_IMAGE = 'ERROR_UPLOAD_IMAGE';
+
+const userToken = `Bearer ${localStorage.getItem('token')}`;
+
+const headers = {
+	'Content-Type': 'application/json',
+	token: userToken,
+};
+export const GetUserProfile = () => async dispatch => {
+	const profileImage =
+		'https://res.cloudinary.com/dby88h516/image/upload/v1580575657/download_mimbni.png';
+	await axios
+		.get(
+			`https://blackninjas-backend-staging.herokuapp.com/api/v1/auth/profile`,
+			{ headers },
+		)
+		.then(res => {
+			const userProfile = res.data;
+
+			(userProfile.data.birthdate = userProfile.data.birthdate
+				? userProfile.data.birthdate.split('T')[0]
+				: ''),
+				(userProfile.data.profileImage = userProfile.data.profileImage
+					? userProfile.data.profileImage
+					: profileImage),
+				dispatch({
+					type: GET_USER_PROFILE,
+					userProfileInfo: userProfile,
+				});
+		});
+};
+
+export const uploadNewImageOnCloud = data => async dispatch => {
+	const loadingImage =
+		'https://res.cloudinary.com/dby88h516/image/upload/v1580716160/barefootnomad/qv7yr2hvce9tjpx0mzh2.gif';
+	const LoaderURL = { profileImage: loadingImage };
+	dispatch({
+		type: HTTP_REQUEST_START,
+		attribute: LoaderURL,
+	});
+
+	await axios
+		.post('https://api.cloudinary.com/v1_1/dby88h516/image/upload', data)
```

Comment:
@shemaeric1234 Kindly use env variables instead of exposing your cloudinary URL.